web-common/src/features/dashboards/dashboard-utils.ts

```diff
@@ -71,12 +71,56 @@ export function prepareSortedQueryBody(
     comparisonTimeRange = undefined;
   }
 
-  if (
-    comparisonTimeRange?.start &&
-    comparisonTimeRange?.end &&
-    !!timeControls.selectedComparisonTimeRange &&
-    sortMeasureName
-  ) {
+  const { sortMeasureName: apiSortName, measures: apiMeasures } =
+    getMeasureNames(
+      sortMeasureName,
+      sortType,
+      !!comparisonTimeRange?.start &&
+        !!comparisonTimeRange?.end &&
+        !!timeControls.selectedComparisonTimeRange,
+    );
+  measures.push(...apiMeasures);
+
+  return {
+    dimensions: [
+      {
+        name: dimensionName,
+      },
+    ],
+    measures,
+    timeRange: {
+      start: timeControls.timeStart,
+      end: timeControls.timeEnd,
+    },
+    comparisonTimeRange,
+    sort: [
+      {
+        desc: !sortAscending,
+        name: apiSortName,
+      },
+    ],
+    where: sanitiseExpression(whereFilterForDimension, undefined),
+    limit: limit.toString(),
+    offset: "0",
+  };
+}
+
+/**
+ * Gets the list of measures for a request by adding comparison based ones if enabled.
+ * Also updates the sort measure name based on sort type.
+ *
+ * @param sortMeasureName
+ * @param sortType
+ * @param hasComparisonTimeRange
+ */
```

Comment:
I think I'm finding this function confusing because it's doing two different things. The comment helps, but maybe it'd be more clear if there were two separate functions: one for listing the measure names, and one for updating the sorted measure. I see that the invocation in `getTDDExportArgs.ts` does not use the latter functionality, which implies that the functionality perhaps shouldn't be coupled.

Reply:
Hmmm It was being used until the last commit. I can split them now.